notifications/bridge.go

```diff
@@ -65,37 +65,41 @@ func (b *bridge) ManifestDeleted(repo distribution.Repository, sm *manifest.Sign
 }
 
 func (b *bridge) LayerPushed(repo distribution.Repository, layer distribution.Layer) error {
-	return b.createLayerEventAndWrite(EventActionPush, repo, layer.Digest())
+	return b.createLayerEventAndWrite(EventActionPush, repo, layer)
 }
 
 func (b *bridge) LayerPulled(repo distribution.Repository, layer distribution.Layer) error {
-	return b.createLayerEventAndWrite(EventActionPull, repo, layer.Digest())
+	return b.createLayerEventAndWrite(EventActionPull, repo, layer)
 }
 
 func (b *bridge) LayerDeleted(repo distribution.Repository, layer distribution.Layer) error {
-	return b.createLayerEventAndWrite(EventActionDelete, repo, layer.Digest())
+	return b.createLayerEventAndWrite(EventActionDelete, repo, layer)
 }
 
 func (b *bridge) createManifestEventAndWrite(action string, repo distribution.Repository, sm *manifest.SignedManifest) error {
-	event, err := b.createManifestEvent(action, repo, sm)
+	manifestEvent, err := b.createManifestEvent(action, repo, sm)
 	if err != nil {
 		return err
 	}
 
-	return b.sink.Write(*event)
+	tagEvent := b.createEvent(EventActionTag)
+	tagEvent.Target = manifestEvent.Target
+
+	return b.sink.Write(*manifestEvent, *tagEvent)
```

Comment:
But, we don't have "tags". They come through the manifest. If they want to see a manifest as a manifest and tag event, they can create that lie in their layer. Otherwise, this doesn't belong in the open source project -- this simply isn't part of the data model yet.